engine/table/src/main/java/io/deephaven/engine/table/impl/locations/local/LocationTableBuilder.java

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.impl.locations.local;
+
+import io.deephaven.engine.table.Table;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+
+/**
+ * Interface for implementations to perform type coercion and specify a table of partition values for observed table
+ * locations. Used by {@link KeyValuePartitionLayout}.
+ */
+public interface LocationTableBuilder {
```

Comment:
I moved this class outside KeyValuePartitionLayout since its now being used in the new URIListKeyValuePartitionLayout. But I might need to move it back inside if enterprise is using it.
Will work with Andy to understand how they use KeyValuePartitionLayout in Enterprise.